hpx/parallel/algorithms/for_each.hpp

```diff
@@ -21,7 +21,7 @@
 #include <hpx/parallel/algorithms/detail/is_negative.hpp>
 #include <hpx/parallel/util/detail/algorithm_result.hpp>
 #include <hpx/parallel/util/foreach_partitioner.hpp>
-#include <hpx/parallel/util/loop.hpp>
+#include <hpx/parallel/util/prefetching.hpp>
```

Comment:
This change is not correct. We should not pull in the `prefetch.hpp` header into the `for_each.hpp`. If a user wants to use prefetching he/she should explicitly include an appropriate header file. Also, for_each depends on `loop.hpp`, so why did you remove it?